pyam/core.py

```diff
@@ -1382,6 +1355,66 @@ def map_regions(self, map_col, agg=None, copy_col=None, fname=None,
         if not inplace:
             return ret
 
+    def subtract(self, other, join_col, new_name, ignore_meta_conflict=False):
+        """
+        Subtract data in ``other`` from ``self``
+
+        Parameters
+        ----------
+        other : pyam.IamDataFrame
+            Object containing timeseries data to subtract
+
+        join_col : str
+            Column to use to subtract the two sets of data (e.g. ``variable``)
+
+        new_name : str
+            String to write in ``join_col`` in the output timeseries e.g.
+            (``variable 1 - variable 2``)
+
+        ignore_meta_conflict : bool, default False
+            If False and ``other`` is an IamDataFrame, raise an error if
+            any meta columns present in ``self`` and ``other`` are not
+            identical.
+
+        Raises
+        ------
+        ValueError
+            The metadata columns in ``self`` and ``other`` are not identical
+
+        ValueError
+            ``self`` or ``other`` contains more than value for ``join_col``
+
+        NotImplementedError
+            The type of ``other`` is not yet supported
+        """
+        if not isinstance(other, IamDataFrame):
+            raise NotImplementedError
+
+        too_many_vals_error = "`{}` contains more than one `{}`"
+        if len(self[join_col].unique()) > 1:
+            raise ValueError(too_many_vals_error.format("self", join_col))
+
+        if len(other[join_col].unique()) > 1:
+            raise ValueError(too_many_vals_error.format("other", join_col))
+
+        out_meta = merge_meta(self.meta, other.meta, ignore_meta_conflict)
+
+        s_data = self.data.copy()
+        o_data = other.data.copy()
+
+        idx = s_data.columns.tolist()
+        idx_tmp = list(set(idx) - set([join_col]) - {"value"})
+
+        s_data = s_data.set_index(idx_tmp).drop(join_col, axis="columns")
+        o_data = o_data.set_index(idx_tmp).drop(join_col, axis="columns")
+
+        res = (s_data - o_data).reset_index()
+        res[join_col] = new_name
+
+        res = IamDataFrame(res)
+        res.meta = out_meta
```

Comment:
the new `meta` table should be down-selected to those models/scenarios that are still in `data` after dropping all `nan`...